garak/generators/openai.py

```diff
@@ -172,17 +174,23 @@ def _call_model(
                 print(msg)
                 return list()
 
-            response = self.generator.create(
-                model=self.name,
-                prompt=prompt,
-                temperature=self.temperature,
-                max_tokens=self.max_tokens,
-                n=generations_this_call,
-                top_p=self.top_p,
-                frequency_penalty=self.frequency_penalty,
-                presence_penalty=self.presence_penalty,
-                stop=self.stop,
-            )
+            create_args = {
+                "model": self.name,
+                "prompt": prompt,
+                "temperature": self.temperature,
+                "max_tokens": self.max_tokens,
+                "n": generations_this_call,
+                "top_p": self.top_p,
+                "frequency_penalty": self.frequency_penalty,
+                "presence_penalty": self.presence_penalty,
+                "stop": self.stop,
+            }
+            create_args = {k: v for k, v in create_args.items() if v is not None}
+            create_args = {
+                k: v for k, v in create_args.items() if k not in self.blocked_params
+            }
```

Comment:
It looks like both `create` methods only differ in the name of the parameter the `prompt` will will be passed as. If both `chat` and `completions` will never expect to pass `None` for a `create` argument, maybe create a shared method:
``` Python
    def _create_params(self)
            create_args = {
                "model": self.name,
                "temperature": self.temperature,
                "max_tokens": self.max_tokens,
                "n": generations_this_call,
                "top_p": self.top_p,
                "frequency_penalty": self.frequency_penalty,
                "presence_penalty": self.presence_penalty,
                "stop": self.stop,
            }
            create_args = {k: v for k, v in create_args.items() if v is not None}
            create_args = {
                k: v for k, v in create_args.items() if k not in self.blocked_params
            }
            return create_args
```

Then here:
```suggestion
            create_args = {"prompt": prompt} | self._create_params()
```

Reply:
excellent, love this